skimage/exposure/_adapthist.py

```diff
@@ -76,23 +76,30 @@ def equalize_adapthist(image, ntiles_x=8, ntiles_y=8, clip_limit=0.01,
     """
     image = img_as_uint(image)
     image = rescale_intensity(image, out_range=(0, NR_OF_GREY - 1))
-    out = _clahe(image, ntiles_x, ntiles_y, clip_limit * nbins, nbins)
-    image[:out.shape[0], :out.shape[1]] = out
+    if ntiles_x or ntiles_y:
+        ntiles_x = ntiles_x or 8
```

Comment:
I actually like it as is, because `0` or `None` are both caught, and that is what I intended.